src/Api/PubnubApi/EndPoint/PubSub/PublishOperation.cs

```diff
@@ -205,67 +205,80 @@ internal void Publish(string channel, object message, bool storeInHistory, int t
-                }, TaskContinuationOptions.ExecuteSynchronously).Wait();
-            }
-
-            if (!string.IsNullOrEmpty(json))
-            {
-                List<object> result = ProcessJsonResponse(requestState, json);
+                    UrlProcessRequest<PNPublishResult>(request, requestState, false).ContinueWith(r =>
+                    {
+                        json = r.Result.Item1;
+                    }, TaskContinuationOptions.ExecuteSynchronously).Wait();
+                }
 
-                if (result != null && result.Count >= 3)
+                if (!string.IsNullOrEmpty(json))
                 {
-                    int publishStatus;
-                    Int32.TryParse(result[0].ToString(), out publishStatus);
-                    if (publishStatus == 1)
+                    List<object> result = ProcessJsonResponse(requestState, json);
+
+                    if (result != null && result.Count >= 3)
                     {
-                        ProcessResponseCallbacks(result, requestState);
+                        int publishStatus;
+                        Int32.TryParse(result[0].ToString(), out publishStatus);
+                        if (publishStatus == 1)
+                        {
+                            ProcessResponseCallbacks(result, requestState);
+                        }
+                        else
+                        {
+                            PNStatusCategory category = PNStatusCategoryHelper.GetPNStatusCategory(400, result[1].ToString());
+                            PNStatus status = new StatusBuilder(config, jsonLibrary).CreateStatusResponse<PNPublishResult>(PNOperationType.PNPublishOperation, category, requestState, 400, new PNException(json));
+                            if (requestState.PubnubCallback != null)
+                            {
+                                requestState.PubnubCallback.OnResponse(default(PNPublishResult), status);
+                            }
+                        }
                     }
                     else
                     {
-                        PNStatusCategory category = PNStatusCategoryHelper.GetPNStatusCategory(400, result[1].ToString());
-                        PNStatus status = new StatusBuilder(config, jsonLibrary).CreateStatusResponse<PNPublishResult>(PNOperationType.PNPublishOperation, category, requestState, 400, new PNException(json));
-                        if (requestState.PubnubCallback != null)
-                        {
-                            requestState.PubnubCallback.OnResponse(default(PNPublishResult), status);
-                        }
+                        ProcessResponseCallbacks(result, requestState);
                     }
                 }
-                else
+
+                CleanUp();
+            }
+            catch (Exception ex)
+            {
+                PNStatusCategory category = PNStatusCategoryHelper.GetPNStatusCategory(400, ex.ToString());
```

Comment:
An Exception should not have the status code of 400

Reply:
fixed it.